### ISSUE TITLE
GDPR compliance

### ISSUE THREAD

**Issue by vertigofactory** [mdsimpson/contact-form-7-to-database-extension#61]
Hi, I was wondering if are you planning to make this plugin GDPR compliant.

You can find more info about the new EU Privacy Law and Wordpress here: https://wordpress.org/news/2018/04/gdpr-compliance-tools-in-wordpress/

Thank you

**Comment by Taranis01** [comment]
why do you think it should be the database Plugin to add GDPR?

**Comment by vertigofactory** [comment]
I think it is necessary because into the new EU privacy law is required that the user must be able to access to all his data. Using your plugin I can store the data provided by the user through a form, so I have to manage this data. As example, if a user asks for his data I should be able to provide him a dump. I discovered an external solution (here: [https://wp-gdpr.eu/add-ons/contact-form-db-7-addon/](https://wp-gdpr.eu/add-ons/contact-form-db-7-addon/)), but it is not free of charge, and the Wordpress team is working on a set of privacy tools for plugin developers, like a common interface, but the plugin should be compatible with this interface, it's not an automatic process.

**Comment by Taranis01** [comment]
ah i understand. However, i'm not the developer ^^

**Comment by luckyday4** [comment]
@vertigofactory It seems that the fonctionality you ask would be native on 4.9.6. 

But this is not the biggest problem with CFDB7. 
The biggest pb is that personal data is not pseudonymised.
As multiple people such as webmaster could access these datas (name, email, phone number) even if he doesn't need to plugin should provide something that allow you tu blur personnal data for non commercial ou Customer relationship users.

**Comment by vertigofactory** [comment]
@luckyday4 exactly, I'm agree with you. about the functionality I asked for: on 4.9.6 there's an interface called privacy tools, a sort of API, but it is required that the plugin make use of this interface (trivially, the wordpress interface cannot automatically know how the plugin saves the data into the db).

**Comment by kunzi** [comment]
I would also like to see a functionality that would automatically delete old data from database after certain amount of years. It seems that personal data should not be stored indefinitely and to manually delete data periodically would be cumbersome.

**Comment by thewebtailors** [comment]
I would love to see an automatic delete after x days feature as well.

**Comment by bobbel78** [comment]
Yes, an automatic deletion of records after x days/months would be great!
